File: api/src/main/java/com/aiko/api/model/Vehicle.java
package com.aiko.api.model;

import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import lombok.Data;

@Data
@Entity
public class Vehicle {
  
  @Id
  @GeneratedValue(strategy=GenerationType.IDENTITY)
  private long id;


  private String name;

  private String model;

  @ManyToOne
  @JoinColumn(name = "line_id")
  private Line line;

  @Embedded
  private VehiclePosition vehiclePosition;
}

File: TestBackEnd.sln/src/TestBackEnd.Domain/Entities/Vehicle.cs
namespace TestBackEnd.Domain.Entities
{
    public class Vehicle : Base
    {
        public string Name { get; private set; }
        
        public string Model { get; private set; }

        public long LineId { get; private set; }
        
        public Line Line { get; set; }

        public VehiclePosition VehiclePosition { get; set; }
        
        public Vehicle() { }

        public Vehicle(string name, string model, long lineId, Line line, VehiclePosition vehiclePosition)
        {
            Name = name;
            Model = model;
            LineId = lineId;
            Line = line;
            VehiclePosition = vehiclePosition;
        }
    }
}